lib/cocoapods/installer/analyzer.rb

```diff
@@ -419,12 +419,30 @@ def generate_target(target_definition, pod_targets)
         end
 
         target.pod_targets = pod_targets.select do |pod_target|
-          pod_target.target_definitions.include?(target_definition)
+          pod_target.target_definitions.include?(target_definition) &&
```

Comment:
this is the important change. the `pod_targets` of an aggregate target used to be pretty much all pods included in the target definition.

This is no longer true, we can have pod targets used only by tests and therefore we have to filter them out.

This line:

```ruby
target_definition.non_inherited_dependencies.map(&:name).include?(pod_target.name)
```

Allows someone to explicitly specify a pod like OCMock which would have been considered a test only pod and it would have failed to be linked.